src/client/components/split-menu/split-menu.tsx

```diff
@@ -94,205 +82,89 @@ export class SplitMenu extends React.Component<SplitMenuProps, SplitMenuState> {
     window.removeEventListener("keydown", this.globalKeyDownListener);
   }
 
-  globalKeyDownListener(e: KeyboardEvent) {
-    if (enterKey(e)) {
-      this.onOkClick();
-    }
-  }
-
-  onSelectGranularity(granularity: Granularity): void {
-    var { split } = this.state;
-    var bucketAction = split.bucketAction as Granularity;
-    this.setState({
-      split: split.changeBucketAction(updateBucketSize(bucketAction, granularity))
-    });
-  }
-
-  onSelectSortOn(sortOn: SortOn): void {
-    var { split } = this.state;
-    var sortAction = split.sortAction;
-    var direction = sortAction ? sortAction.direction : SortExpression.DESCENDING;
-    this.setState({
-      split: split.changeSortExpression(new SortExpression({
-        expression: sortOn.getExpression(),
-        direction
-      }))
-    });
-  }
+  globalKeyDownListener = (e: KeyboardEvent) => enterKey(e) && this.onOkClick();
 
-  onToggleDirection(): void {
-    var { split } = this.state;
-    var { sortAction } = split;
-    this.setState({
-      split: split.changeSortExpression(sortAction.toggleDirection())
-    });
-  }
+  saveGranularity = (granularity: string) => this.setState({ granularity });
 
-  onSelectLimit(limit: number): void {
-    var { essence } = this.props;
-    var { split } = this.state;
-    var { colors } = essence;
+  saveSort = (sort: SortExpression) => this.setState({ sort });
 
-    if (colors) {
-      colors = Colors.fromLimit(colors.dimension, limit);
-    }
+  saveLimit = (limit: number, colors: Colors) => this.setState({ colors, limit });
 
-    this.setState({
-      split: split.changeLimit(limit),
-      colors
-    });
-  }
+  onCancelClick = () => this.props.onClose();
 
   onOkClick() {
-    if (!this.actionEnabled()) return;
-    var { clicker, essence, onClose } = this.props;
-    var { split, colors } = this.state;
-
-    clicker.changeSplits(essence.splits.replace(this.props.split, split), VisStrategy.UnfairGame, colors);
-    onClose();
-  }
-
-  onCancelClick() {
-    var { onClose } = this.props;
+    if (!this.validate()) return;
+    const { split: originalSplit, clicker, essence, onClose } = this.props;
+    const split = this.constructSplitCombine();
+    clicker.changeSplits(essence.splits.replace(originalSplit, split), VisStrategy.UnfairGame, this.state.colors);
     onClose();
   }
 
-  getSortOn(): SortOn {
-    var { essence, dimension } = this.props;
-    var { split } = this.state;
-    return SortOn.fromSortExpression(split.sortAction, essence.dataCube, dimension);
+  private constructGranularity(): Granularity {
+    const { dimension: { kind } } = this.props;
+    const { granularity } = this.state;
+    if (kind === "time") {
+      return new TimeBucketExpression({ duration: Duration.fromJS(granularity) });
+    }
+    if (kind === "number") {
+      return new NumberBucketExpression({ size: parseInt(granularity, 10) });
```

Comment:
10 to może jakaś domyślna wartośc powinna być, po co ją spychąc ?

Reply:
10 tutaj oznacza bazę, parsujemy inta jako decimal